mvrx/src/main/kotlin/com/airbnb/mvrx/ViewLifecycleOwner.kt

```diff
@@ -0,0 +1,105 @@
+package com.airbnb.mvrx
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.lifecycle.OnLifecycleEvent
+
+class ViewLifecycleOwner(private val view: View) : LifecycleOwner {
+
+    private val lifecycleRegistry = LifecycleRegistry(this)
+
+    /**
+     * This is guaranteed to be a MvRxView.
+     */
+    private var _fragment: Fragment? = null
+    val fragment: Fragment?
+        get() {
+            // If a ViewModel is called from onAttachedToWindow, ths Fragment may be attached before the onAttachStateChangeListener is called.
+            if (_fragment == null && view.isAttachedToWindow) {
+                handleAttachedToWindow()
+            }
+            return _fragment
+        }
+    val activity get() = view.context as? AppCompatActivity
+
+    init {
+        view.addOnAttachStateChangeListener(object : View.OnAttachStateChangeListener {
+            override fun onViewDetachedFromWindow(v: View) {
+                lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_STOP)
```

Comment:
onAttach does `onresume`, which is not the corollary to `onStop` - this should be `onPause` if we want to make them symmetric

Reply:
@elihart True but it wouldn't have the correct behavior of locking the lifecycle aware observer. This is one of the challenges of mapping the view lifecycle to the fragment/activity lifecycle